### PR TITLE
PatWork Function and GenIntro chpt

### PR DESCRIPTION
**Pull request by PJPerri**

### CODE REVIEW

File: pretext/Functions/ChapterAssessment.ptx [hunk 1]
```diff
@@ -144,4 +144,89 @@
...
+      - which sum to 12.66666667 and average to be 4.2222222
+      - the square root of which is: 2.054804667
+        """"
+
+      def average( ,  ,  ): 
+
+
+
+
+        return mean
```
Comment (cfcurtis) [inline pretext/Functions/ChapterAssessment.ptx:208]: The rest of the book seems to be 4 spaces indentation, so to be consistent I changed this example as well.

Comment (cfcurtis) [reply pretext/Functions/ChapterAssessment.ptx:208]: Actually, it's inconsistent, whoops.

File: pretext/Functions/Functionscancallotherfunctions.ptx [hunk 2]
```diff
@@ -30,64 +30,110 @@
...
     stack frame for the <c>square</c> function. If it didn't find it there, it would go look in the
     global frame.</p>
-  <!-- <p>Let's use composition to build up a little more useful function. Recall from the
-  dictionaries
-    chapter that we had a two-step process for finding the letter that appears most frequently in a
-    text string:</p>
+  
+  
+  
+  <p>Let's use composition to build up another function: finding the area of <alert>any</alert> regular polygon</p>
```
Comment (cfcurtis) [inline pretext/Functions/Functionscancallotherfunctions.ptx:50]: I like this example, I just latexified it and changed the Python code from camelcase to snake for consistency.

File: pretext/Functions/GlobalVariables.ptx [hunk 3]
```diff
@@ -108,8 +111,18 @@ z = square(x)
 print(z)
     </input>
   </program>
+
+
+  <note>
+
+  <p>
+    <alert>Instructor's Counsel:</alert> <em>Just because you can, doesn't mean you should.</em> Global variables exist,
+  but, as a novice programmer, you really should not use them. </p>
```
Comment (cfcurtis) [inline pretext/Functions/GlobalVariables.ptx:120]: Perhaps adding an exception is not a good idea, but I generally show students to define constants in the global scope.

File: pretext/Functions/intro-Functions.ptx [hunk 4]
```diff
@@ -48,22 +47,29 @@
...
+     </ul>
   </p>
-  <p>In this chapter you will learn about named functions, functions that can be referred to by name
+  <p>In this chapter you will learn about named functions, i.e. functions that are referred to by a name
     when you want to execute them.</p>
   <subsection xml:id="functions_introduction-to-functions_topics">
     <title>Topics</title>
     <p>
-      <ul>
+      <ul marker="">
```
Comment (cfcurtis) [inline pretext/Functions/intro-Functions.ptx:57]: I got rid of the marker="" - if we do it for this one we need to do it for the rest of the chapters.

File: pretext/Functions/intro-Functions.ptx [hunk 5]
```diff
@@ -48,22 +47,29 @@
...
         </li>
         <li>
-          <p>local and global scope</p>
+        <p>Local and global scope and the flow of execution</p>
         </li>
         <li>
-          <p>side effects</p>
+          <p>Side effects</p>
         </li>
       </ul>
```
Comment (cfcurtis) [inline pretext/Functions/intro-Functions.ptx:74]: Removing side effects for now

File: pretext/Functions/intro-Functions.ptx [hunk 6]
```diff
@@ -48,22 +47,29 @@
...
+            </ul>
         </li>
         <li>
-          <p>local and global scope</p>
+        <p>Local and global scope and the flow of execution</p>
         </li>
         <li>
-          <p>side effects</p>
+          <p>Side effects</p>
         </li>
```
Comment (cfcurtis) [inline pretext/Functions/intro-Functions.ptx:73]: Removing mention of side effects to be consistent with sections

File: pretext/GeneralIntro/WPToUnderstandaProgramChangeIt.ptx [hunk 7]
```diff
@@ -1,15 +1,26 @@
...
-    statements just <em>before</em> the code snippet you are trying to understand.</p>
+    diagnostic print statements that print out the types and values of variables. Sometimes referred to as 'echo prints', add these print
+    statements just <em>before</em> and <em>after</em> the code snippet you are trying to understand.</p>
+
+<program language="python">
+    <input>
+      x = 4
+      y = 7
+      print(x, type(x), y, type(y)) # echo print
+      y = x + 0.2 ;print(type(y), "y=",y) # diagnostic print on same line
```
Comment (cfcurtis) [inline pretext/GeneralIntro/WPToUnderstandaProgramChangeIt.ptx:15]: Do we want to show them semicolons? I think this is the only time this is used and it's not very common.

File: pretext/Functions/FlowofExecutionSummary.ptx [hunk 8]
```diff
@@ -33,7 +33,7 @@
   </p>
   <exercise label="question4_11_1">
     <statement>
-      <p>Consider the following Python code. Note that line numbers are included on the left.</p>
```
[the remark this answers is not in the document]
Comment (PJPerri) [reply pretext/Functions/FlowofExecutionSummary.ptx:35]: This I do not know.

File: pretext/GeneralIntro/Comments.ptx [hunk 9]
```diff
@@ -25,6 +25,41 @@ print("Hello, World!")     # Isn't this easy!
...
+      #5 rounding for significant digits
+
+      #6 solutions statement to display the answer 
+
+
+    </input>
+  </program>
+
+  <p>There is one other way in Python that code is sometimes documented, by enclosing portions of the code within triple-quotes, <c>"""   """</c>. 
+  Code within triple-quotes is not executed and can span multiple lines. Strictly speaking, triple-quotes <em>do not create comments</em> because the interpreter can
```
[the remark this answers is not in the document]
Comment (PJPerri) [reply pretext/GeneralIntro/Comments.ptx]: Agree - """ """ are not comments (although Guido V is quoted as saying he's okay to use them as such).

Are commonly used with docstrings (accessed with (help(foo)) and part of the way I teach my students to a) design functions, b) document them, c) feel empowered to look up how other functions work.

Keep as much as you feel is reasonable.

File: pretext/GeneralIntro/MoreAboutPrograms.ptx [hunk 10]
```diff
@@ -5,30 +5,33 @@
...
     </li>
     <li>
-      <title>output</title>
+      <title><term>output</term></title>
       <p>Display data on the screen or send data to a file or other device.</p>
     </li>
     <li>
-      <title>math and logic</title>
-      <p>Perform basic mathematical operations like addition and multiplication and logical
+      <title><term>assignment</term></title>
```
[the remark this answers is not in the document]
Comment (PJPerri) [reply pretext/GeneralIntro/MoreAboutPrograms.ptx:21]: yes

File: pretext/GeneralIntro/WPToUnderstandaProgramChangeIt.ptx (hunk 7 above)
Comment (PJPerri) [reply pretext/GeneralIntro/WPToUnderstandaProgramChangeIt.ptx:15]: Again, an approach I used to use to support novice students with their debugging. 

Not necessary what with VS Codes debugging features.

**Pull request merged**